### PR TITLE
Removing the need to have a public dir in nginx

### PR DESCRIPTION
**Pull request by allandegnan**
Biggest disadvantage of this is you need to warm your cache by requesting the resources from node, but on the bright side it keeps your apps the same between all your envs.  It also means you don't need to worry about your nginx being local to your app.  Do test this, I haven't.  POC obviously

### CODE REVIEW

**Comment by easternbloc** [comment]
Will changes to docker-compose be required?

Note: I've not read the code i'm just saying words.

**Comment by daniel-ac-martin** [comment]
@easternbloc: Looks like you could simplify your Dockerfile and docker-compose.yml. I'd guess it would still work with the extra stuff though.

**Comment by easternbloc** [comment]
@daniel-ac-martin how do you mean simplify your docker-compose? I'm all 👂's

**Comment by daniel-ac-martin** [comment]
@easternbloc: Just that you have some lines relating to the old 'public' volume. So you shouldn't need that any more. (As far as I can tell anyway!)

**Comment by allandegnan** [comment]
Didn't realise you had one.  Updated that too.  Currently, the docker-compose ngind doesn't do caching; thus you'll get none with it.  But that's probably fine?

**Comment by allandegnan** [comment]
Actually, this requires https://github.com/UKHomeOffice/docker-nginx-proxy/pull/61

File: kube/services/k8resources/gro/gro-rc.yaml [hunk 1]
```diff
@@ -42,14 +42,15 @@ spec:
               add_header X-Frame-Options "SAMEORIGIN" always;
               add_header X-Content-Type-Options "nosniff" always;
               add_header X-XSS-Protection "1; mode=block" always;
+              proxy_cache on;
```
Comment (allandegnan) [inline kube/services/k8resources/gro/gro-rc.yaml]: Based on the nginx docs this should be _enough_ to enable caching from the app for hopefully just this location.  It's slightly unclear enough, as you set a zone but I haven't set proxy_cache_path which defines the "zone".  Apparently it defaults though, but to what? O.o

File: kube/services/k8resources/gro/gro-rc.yaml [hunk 2]
```diff
@@ -42,14 +42,15 @@ spec:
...
+              proxy_cache_revalidate on;
+              proxy_cache_lock on;
```
Comment (allandegnan) [inline kube/services/k8resources/gro/gro-rc.yaml]: This makes sure we only send one request back to the app at a time and any subsequent requests should wait on the first and then serve from cache.

**Comment by JoeChapman** [comment]
@allandegnan @easternbloc can I close this?

**Comment by easternbloc** [comment]
it's POC so i'd say for now we can get rid. it'd need opening on a different repo anyway

**Pull request closed without merging**